### PR TITLE
Feature: article update listener

### PR DESCRIPTION
**Pull request by lsh-0**
downloads new articles when article updates are received

### CODE REVIEW

File: src/observer/inc.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                VisibilityTimeout=60, # time allowed to call delete, can be increased
+                WaitTimeSeconds=20 # maximum setting for long polling
+            )
+        if not messages:
+            continue
+        message = messages[0]
+        try:
+            yield message.body
+        finally:
+            # failing while handling a message will see the message deleted regardless
```
Comment (giorgiosironi) [inline src/observer/inc.py]: and is that what you want? It's a lost update

File: src/observer/inc.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# TODO: cache?
+def queue(name):
+    "returns a connection to a named queue"
+    return boto3.resource('sqs').get_queue_by_name(QueueName=name)
+
+def poll(queue_obj):
+    """an infinite poll on the given queue object.
+    blocks for 20 seconds before connection is dropped and re-established"""
+    while True:
+
```
Comment (giorgiosironi) [inline src/observer/inc.py]: I would suggest to insert similar lines two https://github.com/elifesciences/bot-lax-adaptor/blob/develop/src/sqs_adaptor.py#L37 to deal with stopping the process without interrupting the SQS blocking wait, which leads to a message being assigned but ignored and retried later

File: src/observer/ingest_logic.py [hunk 3]
```diff
@@ -356,7 +359,67 @@ def regenerate_all():
     regenerate_many(logic.known_articles())
 
 #
-# upsert from file/dir of article-json
+# upsert article-json from api
+#
+
+if settings.ENV not in ['prod', 'continuumtest']:
```
Comment (giorgiosironi) [inline src/observer/ingest_logic.py]: different in `end2end` vs `prod` is a smell to me, it should be included in the list?

Comment (lsh-0) [inline src/observer/ingest_logic.py]: I can do that. Looking at it now I think hard coding env names is a smell in general. Will switch it to `if settings.DEBUG`

File: src/observer/inc.py (hunk 1 above)
Comment (lsh-0) [inline src/observer/inc.py]: if it fails between the try and finally, it's almost certainly failing in my code somewhere. the database disappearing would be the next likely cause and I haven't ever seen that with psql. mysql, plenty of times ;) perhaps I'll add a retry count - like, attempt to yield this message twice, then fail with delete and error.

**Pull request merged**